### DISCUSSION TITLE
Size calculation method

### DISCUSSION THREAD

**Discussion by og900aero** [sxyazi/yazi#578] [1. Q&A] [answered]
Is it possible to change the size calculation method in some way?

The "problem" is:

yazi
![yazi](https://github.com/sxyazi/yazi/assets/49487376/50675e4e-d680-434d-8a5d-0727c2ca7358)

ls -la, exa, lf file manager show:
![good](https://github.com/sxyazi/yazi/assets/49487376/2fc82ab6-e828-4b9d-9731-7ec6c279bfcc)

The bigger the file size, the bigger the difference.

**Comment by sxyazi** [comment, marked as answer]
No, Yazi's behavior is consistent with `ls -lah`, where `-h` means `--human-readable`, see:

![1](https://github.com/sxyazi/yazi/assets/17523360/1766a49d-5b42-4b19-8027-f648a7c44202)


I guess you are using exa/eza with `alias ls=exa` - their behavior differs from `ls`; this is not a problem with Yazi.

**Reply by ghost** [reply]
By the way, I noticed that Yazi uses incorrect prefixes with the file/directory sizes. Yazi counts in IEC (power of two), but displays JEDEC (power of ten) prefixes. Please let me know if I should open a bug report for tracking purposes.

**Reply by sxyazi** [reply]
Please do. I'll fix it if it's a real bug ;)

**Reply by og900aero** [reply]
You are correct. I use alias for exa.
Although I have to say that it is quite annoying that there are several methods of calculating sizes even under Linux, and so I don't know which one is good, real.

**Comment by og900aero** [comment]
I write the following code into init.lua, and now working great the other method. Thx sxyazi the idea from [c02d695](https://github.com/sxyazi/yazi/commit/c02d69590bda49babe0883ff78341d92e557f381) commit.

```
function ya.readable_size(size)
	local units = { "B", "KB", "MB", "GB", "TB", "PB", "EB" }
	local i = 1
	while size > 1000.0 and i < #units do
		size = size / 1000.0
		i = i + 1
	end
	return string.format("%.f%s", size, units[i])
end
```
